File: src/main/java/com/example/fit_tracker/service/GoalService.java
package com.example.fit_tracker.service;

import com.example.fit_tracker.dto.GoalDTO;
import com.example.fit_tracker.entity.User;

import java.util.List;

public interface GoalService {

    GoalDTO postGoal(GoalDTO dto);
    List<GoalDTO> getGoals();

    GoalDTO updateStatus(Long id);

    Object updateGoal(Long id, GoalDTO dto);

    String deleteGoal(Long id);

    void checkGoalsForUser(User user);
    //Goal Automatization method 
}

File: fit-tracker-frontend/src/api/GoalService.js
// src/api/GoalService.js
import api from './axiosConfig';

export const GoalService = {
    postGoal: async (goalData) => {
        const response = await api.post('/goal', goalData);
        return response.data;
    },
    getGoals: async () => {
        const response = await api.get('/goals');
        return response.data;
    },
    updateGoal: async (id, goalData) => {
        const response = await api.put(`/goal/${id}`, goalData);
        return response.data;
    },
    deleteGoal: async (id) => {
        const response = await api.delete(`/goal/${id}`);
        return response.data;
    },
    updateStatus: async (id) => {
        const response = await api.get(`/goal/status/${id}`);
        return response.data;
    }
};